Orange/widgets/settings.py

```diff
@@ -949,21 +950,12 @@ def filter_value(self, setting, data, domain, attrs, metas):
                 del data[setting.name]
 
     def settings_to_widget(self, widget, domain, *args):
-        context = widget.current_context
-        if context is None:
-            return
-
-        widget.retrieveSpecificSettings()
-
-        for setting, data, instance in \
-                self.provider.traverse_settings(data=context.values, instance=widget):
-            if not isinstance(setting, ContextSetting) or setting.name not in data:
-                continue
-
-            value = self.decode_setting(setting, data[setting.name], domain)
-            setattr(instance, setting.name, value)
-            if hasattr(setting, "selected") and setting.selected in data:
-                setattr(instance, setting.selected, data[setting.selected])
+        # our decode_setting needs domain to be able to restore
```

Comment:
@janezd, @lanzagar 
What do you think? While I admit that this is a bit hackish, it avoids having a copying the whole function to change a single line.

Reply:
I can't believe this is coming from the guy who overruled my excellent modulo trick a few days ago. :)

Is the line in question this line?

``` 
value = self.decode_setting(setting, data[setting.name])
```

Shouldn't it actually be

``` 
value = self.decode_setting(setting, data[setting.name], *args)
```

`ContextHandler.settings_to_widget` accepts `*args` but doesn't use them. I think they were meant to be used exactly in this line.

Would this fix the problem?